inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -248,6 +248,19 @@ void remove_redundant_reorders::run(program& p) {
         if (r_nodes_to_remove.empty())
             continue;
 
+        bool skip = false;
+        for (auto& user : dep.get_users()) {
+            if (user != node && user->can_be_optimized() &&
+             node->get_output_layout().data_padding.lower_size() != (tensor)0 &&
+             node->get_output_layout().data_padding.upper_size() != (tensor)0) {
+                skip = true;
+                break;
+            }
+        }
```

Comment:
Could it be extracted into independent lambda function and reused in all passes?
Let's say:
```
if (node->get_output_layout().data_padding.lower_size() != (tensor)0 &&
    node->get_output_layout().data_padding.upper_size() != (tensor)0 &&
    is_any_user_optimized(..))
        continue;
```